set-timeskip-duration.lua

```diff
@@ -0,0 +1,213 @@
+    end
+    local days = math.floor(ticks/1200)
+    ticks = ticks%1200
+    if days > 0  then
+      print("  " .. tostring(days) .. " day" .. (days == 1 and "" or "s"))
+    end
+    local hours = math.floor(ticks/50)
+    ticks = ticks%50
+    if hours > 0 then
+      print("  " .. tostring(hours) .. " hour" .. (hours == 1 and "" or "s"))
+    end
+    local minutes = math.floor(6*ticks/5)
+    ticks = ticks - (minutes*5)/6
+    if minutes > 0 then
+      print("  " .. tostring(minutes) .. " minute" .. (minutes == 1 and "" or "s"))
+    end
+    local seconds = math.ceil(72*ticks)
+    if seconds > 0 then
+      print("  " .. tostring(seconds) .. " second" .. (seconds == 1 and "" or "s"))
+    end
+  end
+end
+
+function getTargetDate(ticks)
+  local targetYear = tonumber(df.global.cur_year)
+  local targetTick = tonumber(df.global.cur_year_tick)
+  if targetTick + ticks >= 403200 then
+    ticks = ticks - (403200 - targetTick)
+    targetTick = 0
+    targetYear = targetYear + 1
+  end
+  targetYear = targetYear + math.floor(ticks/403200)
+  targetTick = targetTick + (ticks%403200)
+  return targetYear, targetTick
+end
+
+function setTargetDate(scr, ticks) -- df.viewscreen_update_regionst
+  local targetYear, targetTick = getTargetDate(ticks)
+  scr.year = targetYear
+  scr.year_tick = targetTick
+end
+
+local validArgs = utils.invert({
+  'ticks',
+  'years',
+  'months',
+  'days',
+  'hours',
+  'clear',
+  'help'
+})
+local args = utils.processArgs({...}, validArgs)
+
+if args.help then
+  print(usage)
+  return
+end
+
+if args.clear then
+  dfhack.onStateChange.TimeskipMonitor = nil
+  print("Timeskip duration reset to default.")
+  return
+end
+
+local ticks = 0
+
+if args.ticks then
+  if tonumber(args.ticks) and tonumber(args.ticks) >= 0 then
+    ticks = ticks + tonumber(args.ticks)
+  else
+    qerror("Invalid -ticks duration: " .. tostring(args.ticks))
+  end
+end
+
+if args.years then
+  if tonumber(args.years) and tonumber(args.years) >= 0 then
+    ticks = ticks + tonumber(args.years)*403200
+  else
+    qerror("Invalid -years duration: " .. tostring(args.years))
+  end
+end
+
+if args.months then
+  if tonumber(args.months) and tonumber(args.months) >= 0 then
+    ticks = ticks + tonumber(args.months)*33600
+  else
+    qerror("Invalid -months duration: " .. tostring(args.months))
+  end
+end
+
+if args.days then
+  if tonumber(args.days) and tonumber(args.days) >= 0 then
+    ticks = ticks + tonumber(args.days)*1200
+  else
+    qerror("Invalid -days duration: " .. tostring(args.days))
+  end
+end
+
+if args.hours then
+  if tonumber(args.hours) and tonumber(args.hours) >= 0 then
+    ticks = ticks + tonumber(args.hours)*50
+  else
+    qerror("Invalid -hours duration: " .. tostring(args.hours))
+  end
+end
+
+if ticks == 0 and not (args.ticks and tonumber(args.ticks) == 0) then -- permit users to remove the timeskip altogether by setting it to 0 ticks
```

Comment:
`-days 0` appears to trigger the error message (which I tried to use to end world advancement after having passed `-days 8`).

I think this could be simplified by initializing `ticks` to nil earlier (e.g. with just `local ticks`), then reducing this check to just `if ticks == nil`. That way, any valid argument that assigns a number to ticks would avoid triggering an error, and would allow ticks to be 0.

Reply:
This is another good suggestion, will implement.